improver_tests/generate_ancillaries/test_GenerateClearskySolarRadiation.py

```diff
@@ -121,3 +124,97 @@ def test__initialise_input_cubes(
         GenerateClearskySolarRadiation()._initialise_input_cubes(
             target_grid, None, linke_turbidity_on_alternate_grid
         )
+
+
+def test__irradiance_times():
+    """Test returned irradiance times are equispaced time-steps
+    on the specified interval, with spacing temporal_spacing.
+    Where temporal_spacing does not fit evenly into the total interval,
+    a ValueError should be raised."""
+    time = datetime(2022, 1, 1, 0, 0, tzinfo=timezone.utc)
+    accumulation_period = 3  # in hours
+    temporal_spacing = 60  # in mins
+
+    expected_times = [
+        datetime(2021, 12, 31, 21, 0, tzinfo=timezone.utc),
+        datetime(2021, 12, 31, 22, 0, tzinfo=timezone.utc),
+        datetime(2021, 12, 31, 23, 0, tzinfo=timezone.utc),
+        datetime(2022, 1, 1, 0, 0, tzinfo=timezone.utc),
+    ]
+    result = GenerateClearskySolarRadiation()._irradiance_times(
+        time, accumulation_period, temporal_spacing
+    )
+    assert np.all(result == expected_times)
+
+    accumulation_period = 1
+    expected_times = [
+        datetime(2021, 12, 31, 23, 0, tzinfo=timezone.utc),
+        datetime(2022, 1, 1, 0, 0, tzinfo=timezone.utc),
+    ]
+    result = GenerateClearskySolarRadiation()._irradiance_times(
+        time, accumulation_period, temporal_spacing
+    )
+    assert np.all(result == expected_times)
+
+    misaligned_temporal_spacing = 19
+    with pytest.raises(ValueError, match="must be integer multiple"):
+        GenerateClearskySolarRadiation()._irradiance_times(
+            time, accumulation_period, misaligned_temporal_spacing
+        )
+
+
+def test__create_solar_radiation_cube(target_grid):
+
+    solar_radiation_data = np.zeros_like(target_grid.data)
+    time = datetime(2022, 1, 1, 0, 0)
+    accumulation_period = 24
+
+    for at_mean_sea_level in (True, False):
```

Comment:
Suggest moving `at_mean_sea_level` to use pytest parametrise - that would make the two cases tested here easier to debug if the test fails.